sources/packages/backend/apps/workers/src/services/disbursement-schedule/disbursement-schedule.service.ts

```diff
@@ -53,93 +106,179 @@ export class DisbursementScheduleService extends RecordDataModelService<Disburse
+      studentId,
+    );
+  }
 
-    // Associate all the disbursements of the given assessment with MSFAA.
-    disbursements.forEach((disbursement) => {
-      disbursement.msfaaNumber = msfaaNumber;
-      disbursement.modifier = systemUser;
-    });
-    await this.repo.save(disbursements);
+  /**
+   * Checks if the MSFAA number is still valid.
+   * If the study period end date of the previously signed MSFAA is less than 2 years
+   * when compared to current study period start date, then MSFAA is considered to be valid.
+   * @param previousSignedDisbursement previously signed disbursement.
+   * @param firstDisbursement first disbursement of the application.
+   * @returns whether the MSFAA Number is valid.
+   */
+  private async isMSFAANumberStillValid(
+    previousSignedDisbursement: DisbursementSchedule,
+    firstDisbursement: DisbursementSchedule,
+  ): Promise<boolean> {
+    const previousStudyEndDate = new Date(
+      previousSignedDisbursement.studentAssessment.offering.studyEndDate,
+    );
+    const currentStudyStartDate = new Date(
+      firstDisbursement.studentAssessment.offering.studyStartDate,
+    );
+    // Previously signed and completed application offering end date in considered the start date.
+    // Start date of the offering of the current application is considered the end date.
+    return this.msfaaNumberService.isMSFAANumberValid(
+      previousStudyEndDate,
+      currentStudyStartDate,
+    );
+  }
+
+  /**
+   * Create and activate the MSFAA number fetched from SFAS signed MSFAA.
+   * @param studentId student Id.
+   * @param offeringIntensity offering intensity.
+   * @param applicationId application id.
+   * @param auditUserId audit user id.
+   * @param sfasSignedMSFAA SFAS MSFAA number and latest SFAS application end date.
+   * @returns created and activated MSFAA number id.
+   */
+  private async createAndActivateSfasMSFAANumber(
+    student: Student,
+    application: Application,
+    offeringIntensity: OfferingIntensity,
+    auditUserId: number,
+    sfasSignedMSFAA: SFASSignedMSFAA,
+  ): Promise<number | null> {
+    // Create and activate new MSFAA number from the SFAS records.
+    const sfasMSFAANumber = new MSFAANumber();
+    sfasMSFAANumber.msfaaNumber = sfasSignedMSFAA.sfasMSFAANumber;
+    sfasMSFAANumber.dateSigned = getISODateOnlyString(
+      sfasSignedMSFAA.latestSFASApplicationEndDate,
+    );
+    sfasMSFAANumber.dateRequested = null;
+    sfasMSFAANumber.serviceProviderReceivedDate = null;
+    sfasMSFAANumber.referenceApplication = application;
+    sfasMSFAANumber.student = student;
+    sfasMSFAANumber.offeringIntensity = offeringIntensity;
+    sfasMSFAANumber.creator = { id: auditUserId } as User;
+    const createdMSFAANumber = await this.msfaaNumberService.createMSFAA(
+      sfasMSFAANumber,
+    );
+    const activateMSFAANumber =
+      await this.msfaaNumberSharedService.activateMSFAANumber(
```

Comment:
The `activateMSFAANumber` method is invoked using the below.
```ts
existingMSFAA: {
  id: msfaaNumberRecord.id,
  msfaaNumber: msfaaNumberRecord.msfaaNumber,
},
```
In this way my assumption is that it will be executing the code under.
(https://github.com/bcgov/SIMS/blob/c4d4f56aea488bbcb2cacc7af5f0c1a185e189e3/sources/packages/backend/libs/services/src/msfaa-number/msfaa-number-shared.service.ts#L233-L244)
This means the `dateSigned` will be updated to `null`.
Can you check if the assumption is right?
If it is, the E2E tests should be validating this expected DB update.